base/utf16.jl

```diff
@@ -1,10 +1,82 @@
 # This file is a part of Julia. License is MIT: http://julialang.org/license
 
+#=
+@doc """
+@brief      Error messages for Unicode / UTF support
+""" ->
+=#
+#=
+@enum(UTF_ERR,
+      SHORT = 1,
+      CONT,
+      LONG,
+      NOT_LEAD,
+      NOT_TRAIL,
+      NOT_SURROGATE,
+      MISSING_SURROGATE,
+      INVALID,
+      SURROGATE,
+      NULL_TERMINATE,
+)
+=#
+
+const UTF_ERR_SHORT = 1
+const UTF_ERR_CONT  = 2
+const UTF_ERR_LONG  = 3
+const UTF_ERR_NOT_LEAD = 4
+const UTF_ERR_NOT_TRAIL = 5
+const UTF_ERR_NOT_SURROGATE = 6
+const UTF_ERR_MISSING_SURROGATE = 7
+const UTF_ERR_INVALID = 8
+const UTF_ERR_SURROGATE = 9
+const UTF_ERR_NULL_TERMINATE = 10
+const UTF_ERR_MAX = 10
+
+type UnicodeError <: Exception
+    msg::AbstractString
+end
+
+#=
+@doc """
+@brief      Throws UnicodeError with information about the specific error, location, and character
+
+@param[in]  errCode::UTF_ERR
+@param[in]  errPos:: Integer
+@param[in]  errChar::Integer
+
+@throws never returns, always throws UnicodeError
+""" ->
+=#
+function utf_errfunc(errCode::Integer, errPos::Integer, errChar::Integer)
+    const errMsgs = [
```

Comment:
So, all the strings are interpolated, even if they are not used?  Ugh...
In the system we used, we had, in a database, strings like: "Invalid Unicode character at position $1 ($2)".  Being able to reorder (or use more than once) the error values is important because of ordering differences in different languages.
I'm not sure I follow exactly how you recommend doing this...